centreon/www/include/configuration/configObject/contact/DB-Func.php

```diff
@@ -781,221 +784,362 @@ function updateContact_MC($contact_id = null)
-    } else {
-        $ret = $form->getSubmitValue("contact_hostNotifCmds");
-    }
+    $kernel = Kernel::createForWeb();
 
-    if (is_array($ret) || $ret instanceof Countable) {
-        $resultsCount = count($ret);
-    } else {
-        $resultsCount = 0;
+    /** @var Logger $logger */
+    $logger = $kernel->getContainer()->get(Logger::class);
+
+    if ($contactId <= 0) {
+        $logger->error(
+            "contactId must be an integer greater than 0, given value for contactId : {$contactId}",
+            ['file' => __FILE__, 'line' => __LINE__, 'function' => __FUNCTION__, 'contactId' => $contactId]
+        );
+        return false;
     }
 
-    for ($i = 0; $i < $resultsCount; $i++) {
-        $rq = <<<SQL
-                INSERT INTO contact_hostcommands_relation 
-                (contact_contact_id, command_command_id)
-                VALUES
-                (:contact_id, :command_id)
-                SQL;
-        $statement = $pearDB->prepare($rq);
-        $statement->bindValue(':contact_id', (int)$contact_id, \PDO::PARAM_INT);
-        $statement->bindValue(':command_id', (int)$ret[$i], \PDO::PARAM_INT);
-        $dbResult = $statement->execute();
+    try {
+        $query = "DELETE FROM contact_hostcommands_relation WHERE contact_contact_id = :contact_id";
+        $successDelete = $pearDB->executePreparedQuery($pearDB->prepare($query), ['contact_id' => $contactId]);
+
+        if (!$successDelete) {
+            return false;
+        }
+
+        $hostCommandIdsFromForm = $fields["contact_hostNotifCmds"] ?? $form->getSubmitValue("contact_hostNotifCmds");
+
+        if (!is_array($hostCommandIdsFromForm)) {
+            return false;
+        }
+
+        $query = "INSERT INTO contact_hostcommands_relation(contact_contact_id, command_command_id) VALUES(:contact_id, :command_id)";
+        $pdoSth = $pearDB->prepareQuery($query);
+        foreach ($hostCommandIdsFromForm as $hostCommandIdFromForm) {
+            $pearDB->executePreparedQuery(
+                $pdoSth,
+                ['contact_id' => $contactId, 'command_id' => (int)$hostCommandIdFromForm]
+            );
+        }
+        return true;
+    } catch (CentreonDbException $e) {
+        CentreonLog::create()->insertLog(
+            2,
+            "Error while updating the relationship between contacts and host commands | {$e->toJson()}"
+        );
+        return false;
     }
 }
 
-/*
- * For massive change. We just add the new list if the elem doesn't exist yet
- */
 /**
- * @param null $contact_id
- * @param array $ret
+ * @param int $contactId
+ * @return bool
  */
-function updateContactHostCommands_MC($contact_id = null, $ret = array())
+function updateContactHostCommands_MC(int $contactId): bool
 {
     global $form, $pearDB;
-    if (!$contact_id) {
-        return;
+
+    $kernel = Kernel::createForWeb();
+
+    /** @var Logger $logger */
+    $logger = $kernel->getContainer()->get(Logger::class);
```

Comment:
I don't ask for changes on this one because I'm aware it was already like this before your changes but it's important to notice what we are doing here.

We are booting the entire Symfony kernel inside a single update query just to acces the logger service. 
And we are booting it for EVERY legacy query. This operation has a huge cost in term of resources compared to what it brings.